From 6fbc9b1151448f2571eeb52dff0f955f04fe7a0b Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Thu, 30 Jan 2014 12:59:57 +0000
Subject: [PATCH] service: Use pk_client_set_interactive() rather than the
 deprecated PkTask function

The pk_task_set_interactive() function has been removed in PackageKit 0.9.x and
it's fine to call the subclass version on any version of PackageKit.
---
 service/realm-packages.c      | 2 +-
 tests/frob-install-packages.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/service/realm-packages.c b/service/realm-packages.c
index ee61ee2..f9b0877 100644
--- a/service/realm-packages.c
+++ b/service/realm-packages.c
@@ -382,7 +382,7 @@ realm_packages_install_async (const gchar **package_sets,
 	install = g_new0 (InstallClosure, 1);
 	install->task = pk_task_new ();
 	install->automatic = realm_options_automatic_install (options);
-	pk_task_set_interactive (install->task, FALSE);
+	pk_client_set_interactive (PK_CLIENT (install->task), FALSE);
 	pk_client_set_background (PK_CLIENT (install->task), FALSE);
 	install->invocation = invocation ? g_object_ref (invocation) : NULL;
 	g_task_set_task_data (task, install, install_closure_free);
diff --git a/tests/frob-install-packages.c b/tests/frob-install-packages.c
index 8116a6e..61556ec 100644
--- a/tests/frob-install-packages.c
+++ b/tests/frob-install-packages.c
@@ -121,7 +121,7 @@ test_resolve (void)
 	gint i;
 
 	task = pk_task_new ();
-	pk_task_set_interactive (task, FALSE);
+	pk_client_set_interactive (PK_CLIENT (task), FALSE);
 
 	filter = pk_filter_bitfield_from_string ("arch");
 
-- 
1.8.5.3

